packages/flutter/test/cupertino/popup_surface_test.dart

```diff
@@ -0,0 +1,328 @@
+            child: Padding(
+                padding: EdgeInsets.all(32),
+                child: Stack(
+                  fit: StackFit.expand,
+                  children: <Widget>[
+                    CupertinoTheme(
+                      data: CupertinoThemeData(brightness: Brightness.dark),
+                      child: CupertinoPopupSurface(
+                        blurSigma: 0,
+                        isVibrancePainted: false,
+                        child: SizedBox(),
+                      ),
+                    ),
+                  ],
+                )),
+          ),
+        ),
+      );
+
+      // Golden displays the surface color of the CupertinoPopupSurface
+      // in dark mode.
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('cupertinoPopupSurface.surface-color.dark.png'),
+      );
+    });
+    testWidgets('Setting isSurfacePainted to false removes the surface color', (WidgetTester tester) async {
+      await tester.pumpWidget(const CupertinoApp(
+        home: ColoredBox(
+          color: Color(0xff000000),
+          child: Stack(
+            fit: StackFit.expand,
+            children: <Widget>[
+              CupertinoPopupSurface(
+                blurSigma: 0,
+                isVibrancePainted: false,
+                isSurfacePainted: false,
+                child: SizedBox(),
+              ),
+            ],
+          ),
+        ),
+      ));
+
+      // Golden displays a CupertinoPopupSurface with the color removed (should
+      // be empty).
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('cupertinoPopupSurface.surface-color.removed.png'),
+      );
+    });
+  });
+  group('Blur', () {
+    testWidgets('Positive blurSigma applies blur', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const _FilterTest(
+          CupertinoPopupSurface(
+            isSurfacePainted: false,
+            isVibrancePainted: false,
+            blurSigma: 5,
+            child: SizedBox(),
+          ),
+        ),
+      );
+
+      // Golden displays a CupertinoPopupSurface with no vibrance or surface
+      // color, and a blur sigma of 5
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('cupertinoPopupSurface.blur.5.png'),
+      );
+
+      await tester.pumpWidget(
+        const _FilterTest(
+          CupertinoPopupSurface(
+            isSurfacePainted: false,
+            isVibrancePainted: false,
+            child: SizedBox(),
+          ),
+        ),
+      );
+
+      // Golden displays a CupertinoPopupSurface with no vibrance or surface
+      // color, and a blur sigma of 30.
+      await expectLater(
+        find.byType(CupertinoApp),
+        // 30 is the default blur sigma
+        matchesGoldenFile('cupertinoPopupSurface.blur.30.png'),
+      );
+    });
+    testWidgets('Nonpositive blurSigma removes blur', (WidgetTester tester) async {
```

Comment:
Maybe we just assert in the constructor that `blurSigma >= 0`, since negative sigma doesn't make sense anyway.

Reply:
Done!

Reply:
Done! Because I had been doing simulation animations, sometimes the animation would dip into negative territory (e.g. -0.00012), which caused problems where asserts were present. However, this is an edge case and ClampedSimulation also exists, so I think your suggestion is the better way of handling things.